clippy_lints/src/write.rs

```diff
@@ -240,18 +240,22 @@ declare_clippy_lint! {
 
 pub struct Write {
     format_args: FormatArgsStorage,
-    in_debug_impl: bool,
+    debug_impl_depth: u8, // surely nobody will go 256+ `impl Debug`s deep
```

Comment:
We usually use `u32` for a counter like this unless there's a _very_ good reason. In this case there isn't even a benefit since the struct has enough padding to fit a `u32` counter anyways.